scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py

```diff
@@ -288,35 +365,40 @@ def test_hermitian():
     """
     rnd = np.random.RandomState(0)
 
-    sizes = [3, 10, 50]
-    ks = [1, 3, 10, 50]
+    sizes = [3, 12]
+    ks = [1, 2]
     gens = [True, False]
 
-    for s, k, gen in itertools.product(sizes, ks, gens):
-        if k > s:
-            continue
-
+    for s, k, gen, dh, dx, db in (
+        itertools.product(sizes, ks, gens, gens, gens, gens)
+    ):
         H = rnd.random((s, s)) + 1.j * rnd.random((s, s))
         H = 10 * np.eye(s) + H + H.T.conj()
+        H = H.astype(np.complex128) if dh else H.astype(np.complex64)
```

Comment:
Now that we have a parametrized `test_dtypes`, do we still need (or want) the changes in `test_diagonal`?

Reply:
> Now that we have a parametrized `test_dtypes`, do we still need (or want) the changes in `test_diagonal`?

The only change in `test_diagonal_data_types` is to make the numbers smaller so that the tests runs much faster - currently one of the slowest in scipy.

Reply:
That's not the only change - you're also adding more parameters to the itertools.product and switching to complex casts based on them or not. Since this PR is going to be squash-merged, I'd like to keep unrelated changes to a minimum. We can decrease the numbers if you want, but I'd like to leave the test untouched if we can.

Reply:
The code quoted above adding `np.complex128` is in `test_hermitian`. There are no changes in the `test_diagonal` function. There is `test_diagonal_data_types`, but the only change in `test_diagonal_data_types` is to make the numbers smaller so that the tests runs much faster - currently one of the slowest in scipy.